### PR TITLE
push Docker images to GHCR.io

### PR DESCRIPTION
**Pull request by epwalsh**
#162 Added a Dockerfile, this PR pushes images built from that Dockerfile to GitHub's [container registry](https://github.com/allenai/tango/pkgs/container/tango) on every commit and release, and adds information to the README and docs about using the Dockerfile and pre-built images for users' own projects.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,6 +116,21 @@ tango info
 
 to check your installation.
 
+### Docker image
+
+You can build a Docker image suitable for tango projects by using [the official Dockerfile](https://github.com/allenai/tango/blob/main/Dockerfile) as a starting point for your own Dockerfile, or you can simply use one of our [prebuilt images](https://github.com/allenai/tango/pkgs/container/tango) as a base image in your Dockerfile. For example:
+
+```Dockerfile
+FROM ghcr.io/allenai/tango:cuda11.5.1
```
Comment (schmmd) [inline README.md:124]: Do you want to include a version of tango in the image id?

File: README.md [hunk 2]
```diff
@@ -116,6 +116,21 @@ tango info
...
+
+# Install your project's additional requirements.
+COPY requirements.txt .
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Install source code.
```
Comment (schmmd) [inline README.md:130]: Maybe add a comment that people should have a `.dockerignore` to exclude unwanted files.

**Review by schmmd — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (epwalsh) [reply README.md:133]: Done: e208bcb

File: README.md (hunk 1 above)
Comment (epwalsh) [reply README.md:127]: This better? e208bcb

Comment (epwalsh) [reply README.md:127]: I didn't want to hard-code a version since that would quickly go out-of-date.

Comment (epwalsh) [reply README.md:127]: We won't have a versioned tag (like `tango:0.4.1-cuda11.5.1`) until the next release.

**Pull request merged**